src/SFA.DAS.Employer.PR.Web/Controllers/SelectLegalEntityController.cs

```diff
@@ -84,4 +87,14 @@ private void SetSessionModel(SelectLegalEntitiesViewModel currentModel)
 
         return model;
     }
+
+    private async Task SetSessionForSelectionMade(string employerAccountId, long legalEntityId,
+        CancellationToken cancellationToken)
+    {
+        var response = await _outerApiClient.GetAccountLegalEntities(employerAccountId, cancellationToken);
```

Comment:
As discussed this can be held in session, so we have to retrieve this information only once for the journey.

Reply:
Also once you store this, the legal entity count will be made redundant

Reply:
yes I've now done all that